50.IESProfileTest/compute/cdc.comp

```diff
@@ -0,0 +1,189 @@
+{
+	uint anglesLength = vAngles.length();
+    int idx = -1;
+	
+    for (int i = 0; i < anglesLength; ++i) {
+        if (vAngles[i] > angle) {
+            idx = i;
+            break;
+        }
+    }
+	
+    return ivec2(idx, anglesLength);
+}
+
+ivec2 implGetHUpperBoundIdx(double angle)
+{
+	uint anglesLength = hAngles.length();
+    int idx = -1;
+	
+    for (int i = 0; i < anglesLength; ++i) {
+        if (hAngles[i] > angle) {
+            idx = i;
+            break;
+        }
+    }
+	
+    return ivec2(idx, anglesLength);
+}
+
+int getVLowerBoundIdx(double angle) 
+{
+    return max(implGetVUpperBoundIdx(angle).x - 1, 0);
+}
+
+int getHLowerBoundIdx(double angle) 
+{
+    return max(implGetHUpperBoundIdx(angle).x - 1, 0);
+}
+
+int getVUpperBoundIdx(double angle)
+{
+	ivec2 returnValue = implGetVUpperBoundIdx(angle);
+    return min(returnValue.x, returnValue.y - 1);
+}
+
+int getHUpperBoundIdx(double angle)
+{
+	ivec2 returnValue = implGetHUpperBoundIdx(angle);
+    return min(returnValue.x, returnValue.y - 1);
+}
+
+double getValue(int i, int j)
+{
+	return data[vAngles.length() * i + j];
+}
+
+double sampleI(double vAngle, double hAngle)
+{
+	double vABack = vAngles[vAngles.length() - 1];
+	double hABack = hAngles[hAngles.length() - 1];
+
+	if (vAngle > vABack)
+		return 0.0;
+	
+	hAngle = hABack == 0.0 ? 0.0 : fmod(hAngle, hABack);
+
+	// bilinear interpolation
+	int j0 = getVLowerBoundIdx(vAngle);
+	int j1 = getVUpperBoundIdx(vAngle);
+	int i0 = getHLowerBoundIdx(hAngle);
+	int i1 = getHUpperBoundIdx(hAngle);
+	
+	double uReciprocal = i1 == i0 ? 1.0 : 1.0 / (hAngles[i1] - hAngles[i0]);
+	double vReciprocal = j1 == j0 ? 1.0 : 1.0 / (vAngles[j1] - vAngles[j0]);
+	
+	double u = (hAngle - hAngles[i0]) * uReciprocal;
+	double v = (vAngle - vAngles[j0]) * vReciprocal;
+	
+	double s0 = getValue(i0, j0) * (1.0 - v) + getValue(i0, j1) * (v);
+	double s1 = getValue(i1, j0) * (1.0 - v) + getValue(i1, j1) * (v);
+	
+	return s0 * (1.0 - u) + s1 * u;
+}
+
+void main()
+{
+	const double VERTICAL_INVERSE = 1.0 / TEXTURE_SIZE;
+	const double HORIZONTAL_INVERSE = 1.0 / TEXTURE_SIZE;
+
+	const ivec2 pixelCoordinates = ivec2(gl_GlobalInvocationID.xy);
+	const ivec2 destinationSize = imageSize(outIESCandelaImage);
+	
+	if (all(lessThan(pixelCoordinates, destinationSize)))
+	{
+		const vec2 uv = vec2(pixelCoordinates.x * VERTICAL_INVERSE, pixelCoordinates.y * HORIZONTAL_INVERSE);
+		vec3 direction = octahedronUVToDir(uv);
+		vec2 sphericalCoordinates = sphericalDirToAngles(direction); // third radius spherical compoment is normalized and skipped
+		
+		double intensity = sampleI(degrees(sphericalCoordinates.x), degrees(sphericalCoordinates.y));
+		vec4 value = vec4(intensity / pc.maxIValue, 0, 0, 0);
+		
+		imageStore(outIESCandelaImage, pixelCoordinates, value);
+		imageStore(outSphericalCoordinatesImage, pixelCoordinates, vec4(sphericalCoordinates.xy, 0, 1));
+		imageStore(outOUVProjectionDirectionImage, pixelCoordinates, vec4(direction.xyz, 1));
```

Comment:
something also useful that I'd do is take the IES data grid and scatter (you take raw spherical coordinate & value then figure out the destination pixel and write it there) it to an image as a "point cloud", see if there's some discrepancies